packages/shopify-app-session-storage-drizzle/src/adapters/drizzle-pg.adapter.ts

```diff
@@ -0,0 +1,139 @@
+import {Session} from '@shopify/shopify-api';
+import {SessionStorage} from '@shopify/shopify-app-session-storage';
+import {
+  InferInsertModel,
+  InferSelectModel,
+  desc,
+  eq,
+  inArray,
+} from 'drizzle-orm';
+import {
+  PgDatabase,
+  QueryResultHKT as PgDatabaseQueryResultHKT,
+} from 'drizzle-orm/pg-core';
+
+import {PgSessionTable} from '../schemas/pg.schema';
+
+export class DrizzlePgAdapter<
+  T extends PgDatabase<PgDatabaseQueryResultHKT, TSchema>,
+  TSchema extends {[key: string]: unknown} = {[key: string]: unknown},
```

Comment:
Should this extend the schema type instead?
```suggestion
  TSchema extends PgSessionTable,
```

If yes, we should probably type `sessionTable` as `TSchema` in the constructor, right?

Reply:
No, this is not possible with Drizzle as I know